src/config/validation.rs

```diff
@@ -159,12 +160,16 @@ pub fn check_outputs(config: &ConfigBuilder) -> Result<(), Vec<String>> {
     }
 
     for (key, transform) in config.transforms.iter() {
-        let definition = schema::Definition::empty();
+        // use the most general definition possible, since the real value isn't known yet.
+        let definition = Definition::any();
```

Comment:
I prefer to keep the `schema` module in-line, as it is more self-describing to readers as just seeing the type `Definition`.

Alternatively, we could rename the type to `SchemaDefinition`, but that dings the Clippy lint of repeating module names, which exists specifically because you can just reference `schema::Definition` to get the same result in terms of readability of the code.